rhombus/private/function.rkt

```diff
@@ -57,12 +58,30 @@
     (raise-argument-error* who rhombus-realm "Function" proc)))
 
 (define (check-list who l)
-  (unless (list? l)
+  (unless (treelist? l)
     (raise-argument-error* who rhombus-realm "List" l)))
 
+(define (check-proc-arity who fn n)
+  (unless (procedure-arity-includes? fn n)
+    (raise-arguments-error* who rhombus-realm
+                            (string-append
+                             "map: argument mismatch;\n"
+                             " the given procedure's expected number of arguments does not match the given number of lists\n")
+                            "given procedure" fn
+                            "expected" (procedure-arity fn)
```

Comment:
These numbers should be converted to strings and then wrapped as `unquoted-printing-string`s.

Reply:
HF